src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```diff
@@ -263,6 +318,42 @@ private Listing<Job> mapToJobListing(final List<String> stdOut) {
     }
 
     private boolean jobNotFoundByIdError(final CommandResult result) {
-        return result.getStdErr().toString().equals(jobIdNotFoundMessage);
+        if (!result.getStdErr().isEmpty()) {
+            return result.getStdErr().get(0).equals(jobIdNotFoundMessage);
+        }
+        return false;
+    }
+
+    /**
+     * This method creates the structure of the executed job deletion command based on the passed options.
+     *
+     * @param filter User-defined conditions for searching for deleted jobs.
+     * @return The structure of an executable command.
+     */
+    private String[] makeScancelCommand(final DeleteJobFilter filter) {
+        final Context context = new Context();
+        context.setVariable(JOB_FILTER, filter);
+        return commandCompiler.compileCommand(getProviderType(), SCANCEL_COMMAND, context);
+    }
+
+    private List<Long> parseDeletedJobId(final List<String> stdOut) {
+        final Map<String, String> errorDeletingJobs = stdOut.stream()
+                .filter((s) -> s.startsWith(KILL_JOB_ERROR_PREFIX))
+                .collect(Collectors.toMap(
+                            (s) -> s.substring(ERROR_JOB_ID_POSITION, s.indexOf(COLON, ERROR_JOB_ID_POSITION)),
+                            Function.identity()));
+
+        final List<Long> deletedJobs = stdOut.stream()
+                .filter((s) -> s.startsWith(START_TERMINATING_JOB_PREFIX))
+                .map((s) -> s.substring(JOB_ID_START_POSITION))
+                .filter((id) -> !errorDeletingJobs.containsKey(id))
+                .map(Long::valueOf)
+                .collect(Collectors.toList());
+
+        if (deletedJobs.isEmpty()) {
```

Comment:
better to move this logic to method that call `parseDeletedJobId` because this method (as it expecting from naming) should only **parse deleted job**

Reply:
moved this logic into the calling method